### ISSUE TITLE
GBM models in h2o

### ISSUE THREAD

**Issue by zizaozi** [h2oai/h2o-tutorials#76]
I am new to h2o. I ran a simulation study where I want to recover a simple tree. I have two predictors: x1 and x2 where x2 is categorical with 26 levels (a, b, ..., z). A simple tree is as follows: 

_If x1 > 0.5 or x2 is from one of the following characters ['w','a','n','g','s','h','e'], the I will assign y a yes, otherwise no._

I am trying to use GBM models to recover this tree. The code is as follows:

`
####################### Start of code #########################
import pandas as pd
import numpy as np
import random
import h2o
from h2o.estimators.gbm import H2OGradientBoostingEstimator
h2o.init()             

random.seed(7)
x1 = [random.gauss(mu = 0, sigma=2) for a in range(52)]
x2 = np.repeat(list(string.ascii_lowercase),2)
y = ['Yes' if x1[a] > 0.5 or x2[a] in ['w','a','n','g','s','h','e'] else 'No' for a in range(52)]

h2o_model = H2OGradientBoostingEstimator(ntrees=5, max_depth=3, seed=10, sample_rate=1, distribution='bernoulli')
X2 = h2o.H2OFrame(pd.DataFrame({'x1':x1,'x2':x2, 'y2':y2}))
X2['y2'] = X2['y2'].asfactor()
h2o_model.train(y='y2', training_frame = X2)

h2o_model.predict(h2o.H2OFrame(pd.DataFrame({'x1':x1,'x2':x2})))
#################### End of code ##############################
`

However, the output of the last line is below (shows only the first a bunch of rows):

   predict        No       Yes
**0       No  0.374969  0.625031**
1      Yes  0.283827  0.716173
2       No  0.693166  0.306834
3       No  0.693166  0.306834
4       No  0.650580  0.349420
5       No  0.650580  0.349420
6      Yes  0.286945  0.713055
7      Yes  0.283955  0.716045
8      Yes  0.286945  0.713055
9      Yes  0.283955  0.716045
10     Yes  0.328816  0.671184
11      No  0.598123  0.401877
**12      No  0.374969  0.625031**
13     Yes  0.283827  0.716173
14     Yes  0.286945  0.713055
15     Yes  0.286945  0.713055
16      No  0.651392  0.348608
...

It looks like that for the 0th and the 12th line, the prediction is not in line with the probabilities. For the first line, the probability of getting 'Yes' is more than that of getting a 'No'. Why the prediction is 'No' for this example? There are in total 6 rows like them.  

Anyone has any idea about why? Thanks very much.
